### PR TITLE
Implement All Required Endpoints

### PR DESCRIPTION
**Pull request by Lekipising**
- Get invoices
- Get Payouts
- Get Payment Accounts
- Get Total Paid Invoices
- Get the Last 7 Days' Transactional Value
- Get the Last 30 Days' Transactional Value
- Get Average Daily Transactional Value
- Get Merchant Balance
- Get Projected Balance for a Merchant

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -47,7 +42,16 @@
 			<artifactId>postgresql</artifactId>
 			<scope>runtime</scope>
     	</dependency>
-		        <dependency>
+		<dependency>
+            <groupId>com.h2database</groupId>
```
Comment (filschristian) [inline pom.xml:46]: Are you going to need this?

File: src/main/java/com/irembo/portal/controller/PaymentChargeItemController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.irembo.portal.model.PaymentChargeItem;
+import com.irembo.portal.repository.PaymentChargeItemRepository;
+
+@RestController
+@RequestMapping("/api")
```
Comment (filschristian) [inline src/main/java/com/irembo/portal/controller/PaymentChargeItemController.java:14]: api/payment-charger-something..... I am following your conventions here

File: src/main/java/com/irembo/portal/controller/PaymentChargeItemController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+public class PaymentChargeItemController {
+    private final PaymentChargeItemRepository paymentChargeItemRepository;
+
+    @Autowired
+    public PaymentChargeItemController(PaymentChargeItemRepository paymentChargeItemRepository) {
+        this.paymentChargeItemRepository = paymentChargeItemRepository;
+    }
+
+    @GetMapping("/payment-charge-items")
```
Comment (filschristian) [inline src/main/java/com/irembo/portal/controller/PaymentChargeItemController.java:23]: You can move the /payment..... up

File: src/main/java/com/irembo/portal/controller/PaymentInvoiceExternalNotificationController.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+@RequestMapping("/api")
+public class PaymentInvoiceExternalNotificationController {
+    private final PaymentInvoiceExternalNotificationRepository paymentInvoiceExternalNotificationRepository;
+
+    @Autowired
+    public PaymentInvoiceExternalNotificationController(PaymentInvoiceExternalNotificationRepository paymentInvoiceExternalNotificationRepository) {
+        this.paymentInvoiceExternalNotificationRepository = paymentInvoiceExternalNotificationRepository;
+    }
+
+    @GetMapping("/payment-invoice-external-notifications")
```
Comment (filschristian) [inline src/main/java/com/irembo/portal/controller/PaymentInvoiceExternalNotificationController.java:23]: same here

File: src/main/java/com/irembo/portal/controller/PaymentInvoiceItemController.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+@RequestMapping("/api")
+public class PaymentInvoiceItemController {
+    private final PaymentInvoiceItemRepository paymentInvoiceItemRepository;
+
+    @Autowired
+    public PaymentInvoiceItemController(PaymentInvoiceItemRepository paymentInvoiceItemRepository) {
+        this.paymentInvoiceItemRepository = paymentInvoiceItemRepository;
+    }
+
+    @GetMapping("/payment-invoice-items")
```
Comment (filschristian) [inline src/main/java/com/irembo/portal/controller/PaymentInvoiceItemController.java:23]: same here

**Review by filschristian — CHANGES_REQUESTED** [review]
Can you put the entities in their own folder. Under model
For a better design.

Also, are you going to need all these endpoints? We had agreed to only focus on the work that you need

File: pom.xml (hunk 1 above)
Comment (Lekipising) [reply pom.xml:46]: No. I will remove

**Comment by Lekipising** [comment]
> 

Not all of them, but some of them will support my implementation, e.g used to create data I will need while creating invoices. I will remove them as I continue to work on the endpoints I need

**Comment by Lekipising** [comment]
> Can you put the entities in their own folder. Under model For a better design.
> 
> Also, are you going to need all these endpoints? We had agreed to only focus on the work that you need

All entities are in folder `model`. Please check

Not all of them, but some of them will support my implementation, e.g used to create data I will need while creating invoices. I will remove them as I continue to work on the endpoints I need

**Pull request merged**